feat(rpc): enhance RPC methods error handling

silva-fj opened the pull request:
## Summary
- Add comprehensive error handling infrastructure with detailed error reporting
- Implement centralized validation helpers for consistent input validation
- Improve all RPC methods with better error messages and user-friendly suggestions

silva-fj committed:
feat(rpc): add detailed error handling infrastructure

- Add DetailedError struct with request IDs and field details
- Add validation helpers for common input validation
- Add configuration module for chain constants

silva-fj committed:
refactor(rpc): remove request_id from DetailedError

- Remove auto-generated request_id field from ErrorDetails struct
- Remove UUID dependency and imports
- Simplify error structure to avoid confusion with client request IDs

silva-fj committed:
refactor(rpc): organize error codes by category

silva-fj committed:
feat(rpc): improve error handling in RPC methods

- Use DetailedError for all validation errors
- Add comprehensive input validation with helpers
- Provide user-friendly error messages and suggestions
- Consistent error propagation across all methods

silva-fj committed:
refactor(native-task): improve error handling in native task handler

- Add detailed error types for different failure scenarios
- Improve error propagation from native tasks
- Better error context for debugging

silva-fj committed:
chore(deps): add email_address dependency for validation

silva-fj committed:
Merge branch 'dev' into p-1660-check-all-rpc-methods-error-handling

Resolved conflicts:
- native-task-handler/src/lib.rs: Keep specific error types while adapting to new InternalError format
- rpc-server/src/error_code.rs: Include both INTERNAL_ERROR_CODE and new specific error codes
- rpc-server/src/methods/omni/common.rs: Handle InternalError with Option<String> parameter

silva-fj committed:
feat(rpc): update supported EVM chains

- Add Sepolia, Arbitrum Sepolia, Optimism Sepolia testnets
- Add Polygon Mumbai, Base Sepolia, Local Anvil
- Remove deprecated Goerli testnets and BSC

silva-fj committed:
test(rpc): update chain validation tests for new chains

silva-fj committed:
feat(aa-client): add gas/retry configs for new chains

silva-fj committed:
refactor: remove unused error codes and fix duplicates

- Remove unused error codes
- Consolidate duplicate error code definitions between detailed_error.rs and error_code.rs
- Export error codes from error_code.rs and update imports in common.rs

BillyWooo commented on tee-worker/omni-executor/rpc-server/src/config/chains.rs:
```diff
@@ -0,0 +1,17 @@
+// Supported EVM-compatible chains
+pub const SUPPORTED_EVM_CHAINS: &[u32] = &[
```
why here is no `bsc` and `bsc-testnet` ?

BillyWooo reviewed (commented):
Thank you very much. It looks very good for me. 
I left one question there. 

@kziemianek can you also take a look?

silva-fj committed:
Merge remote-tracking branch 'origin/dev' into p-1660-check-all-rpc-methods-error-handling

silva-fj committed:
feat(rpc): add BSC mainnet and testnet to supported chains

silva-fj committed:
refactor(rpc): update submit_user_op_with_auth to use DetailedError pattern

- Replace all PumpxRpcError::from_error_code with DetailedError
- Add validation helpers for chain_id, wallet_index, and user_operations
- Provide detailed error messages with field context and suggestions
- Fix remaining inconsistent error handling patterns

silva-fj committed:
fix: update validation tests for BSC chain support

- Remove chain ID 56 from invalid test assertions
- Add BSC mainnet (56) and testnet (97) to valid test assertions
- Aligns tests with recently added BSC support in SUPPORTED_EVM_CHAINS

silva-fj committed:
Merge branch 'dev' into p-1660-check-all-rpc-methods-error-handling

kziemianek commented on tee-worker/omni-executor/rpc-server/src/methods/omni/common.rs:
```diff
@@ -63,6 +70,27 @@ impl From<PumpxRpcError> for ErrorObjectOwned {
+			message: error.message,
+			data: Some(PumpxRpcErrorData {
+				backend_response: Some(PumpxRpcErrorBackendResponse {
+					code: error.code,
```
Why `backend_response` is always present ?
It means it will be also present for errors occured in omni-executor and might be not related to pumpx backend - am I right ?

silva-fj replied:
`backend_response` should be `None` by default, I missed that. It will only be present when is `Some`. I'll fix it

kziemianek commented on tee-worker/omni-executor/rpc-server/src/detailed_error.rs:
```diff
@@ -0,0 +1,197 @@
+			.with_field("chain_id")
+			.with_received(chain_id.to_string())
+			.with_expected(format!("One of: {:?}", supported))
+			.with_suggestion("Please use a supported chain ID")
```
There is already `INVALID_CHAIN_ID_CODE` , why we also need `UNSUPPORTED_CHAIN_CODE` ?

BillyWooo reviewed (approved):
LGTM. 
@silva-fj  Please also check and apply changes to `get_hyperliquid_signature_data.rs`. Thank you.

kziemianek commented on tee-worker/omni-executor/rpc-server/src/detailed_error.rs:
```diff
@@ -0,0 +1,197 @@
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ErrorDetails {
+	#[serde(skip_serializing_if = "Option::is_none")]
+	pub field: Option<String>,
```
What's the meaning of the `field` here and how it should be used ? 
Why in some cases it's populated with `operation` ?

silva-fj replied:
The `field` is meant for invalid field params. I'll remove the usage with operation

silva-fj committed:
feat(rpc): add reason field to DetailedError for better error context

silva-fj committed:
refactor(rpc): use with_reason for internal errors in validation helpers

silva-fj committed:
refactor(rpc): migrate common error handling to DetailedError pattern

silva-fj committed:
refactor(rpc): migrate omni methods to use with_reason for errors

silva-fj committed:
refactor(rpc): update submit and export methods error handling

silva-fj committed:
refactor(rpc): complete migration to with_reason for remaining methods

silva-fj committed:
fix(rpc): remove inappropriate with_field from parse errors

silva-fj committed:
refactor(rpc): migrate get_hyperliquid_signature_data to DetailedError pattern

- Replace ErrorCode with DetailedError using to_error_object()
- Use validate_ethereum_address helper for address validation
- Add structured error context with field, reason, and suggestion
- Improve error messages for better debugging and user guidance

silva-fj committed:
chore: remove unused validate_hex_string function and add missing import

- Remove dead_code validate_hex_string from validation_helpers
- Add FromStr import for tests in get_hyperliquid_signature_data

silva-fj committed:
Merge branch 'dev' into p-1660-check-all-rpc-methods-error-handling

Kailai-Wang commented on tee-worker/omni-executor/rpc-server/src/methods/omni/estimate_user_op_gas.rs:
```diff
@@ -74,16 +73,30 @@ pub fn register_estimate_user_op_gas<
 
-			let account_id =
-				decode_account_id(&params.omni_account).map_err(PumpxRpcError::from_error_code)?;
+			let account_bytes = validate_omni_account_hex(&params.omni_account, "omni_account")
+				.map_err(PumpxRpcError::from)?;
+			validate_omni_account_length(&account_bytes, "omni_account")
+				.map_err(PumpxRpcError::from)?;
+			let account_id = AccountId::decode(&mut &account_bytes[..]).map_err(|e| {
+				PumpxRpcError::from(DetailedError::account_parse_error(
+					&params.omni_account,
+					&e.to_string(),
+				))
+			})?;
```
A bit too verbose to me but it's fine as it returns finer error message

Kailai-Wang reviewed (approved):
Looks good to me!

silva-fj committed:
Merge branch 'dev' into p-1660-check-all-rpc-methods-error-handling
